src/rules/noInferrableTypesRule.ts

```diff
@@ -104,12 +107,12 @@ function shouldCheck(node: ts.Node, { ignoreParameters, ignoreProperties }: Opti
     }
 }
 
-function typeIsInferrable(type: ts.SyntaxKind, initializer: ts.SyntaxKind): boolean {
+function typeIsInferrable(type: ts.SyntaxKind, initializer: ts.SyntaxKind, initializerText: string): boolean {
     switch (type) {
         case ts.SyntaxKind.BooleanKeyword:
             return initializer === ts.SyntaxKind.TrueKeyword || initializer === ts.SyntaxKind.FalseKeyword;
         case ts.SyntaxKind.NumberKeyword:
-            return initializer === ts.SyntaxKind.NumericLiteral;
+            return initializer === ts.SyntaxKind.NumericLiteral || initializerText.indexOf(INFINITY_TEXT) !== -1;
```

Comment:
A little helper to avoid `getText()`, handle negative numbers, `Infinity`, `-Infinity`, and `NaN`
```ts
function isNumeric(node: ts.Expression) {
    while (isPrefixUnaryExpression(node) &&
           (node.operator === ts.SyntaxKind.PlusToken || node.operator === ts.SyntaxKind.MinusToken)) {
        node = node.operand;
    }
    return node.kind === ts.SyntaxKind.NumericLiteral ||
        isIdentifier(node) && (node.text === "NaN" || node.text === "Infinity");
}
```